node/src/components/consensus/era_supervisor.rs

```diff
@@ -1254,52 +1240,36 @@ async fn check_deploys_for_replay_in_previous_eras_and_validate_block<REv, I>(
     proposed_block_era_id: EraId,
     sender: I,
     proposed_block: ProposedBlock<ClContext>,
-) -> Result<Event<I>, ReplayCheckAndValidateBlockError>
+) -> Event<I>
 where
     REv: From<BlockValidationRequest<BlockPayload, I>> + From<StorageRequest>,
     I: Clone + Send + 'static,
 {
     for deploy_hash in proposed_block.value().deploys_and_transfers_iter() {
-        let execution_results = match effect_builder
-            .get_deploy_and_metadata_from_storage(*deploy_hash)
+        let block_header = match effect_builder
+            .get_block_header_for_deploy_from_storage(*deploy_hash)
             .await
         {
             None => continue,
-            Some((_, DeployMetadata { execution_results })) => execution_results,
+            Some(header) => header,
         };
-        // We have found the deploy in the database.  If it was from a previous era, it was a
-        // replay attack.  Get the block header for that deploy to check if it is provably a replay
+        // We have found the deploy in the database. If it was from a previous era, it was a
+        // replay attack.
+        //
+        // If the deploy was included in a block which is from before the current era_id
+        // then this must have been a replay attack.
```

Comment:
Aren't these two paragraphs saying effectively the same thing?

Reply:
Ah, right. I tried joining the old comments, but I see I missed some repetition :grimacing: